File: account-service/src/main/java/com/vegesoft/moneytracker/account/configuration/WebClientConfiguration.java
package com.vegesoft.moneytracker.account.configuration;

import com.vegesoft.moneytracker.account.configuration.properties.WebClientProperties;
import io.netty.channel.ChannelOption;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.http.HttpHeaders;
import org.springframework.http.MediaType;
import org.springframework.http.client.reactive.ReactorClientHttpConnector;
import org.springframework.web.reactive.function.client.WebClient;

@Configuration
public class WebClientConfiguration {

    private final WebClientProperties webClientProperties;

    public WebClientConfiguration(final WebClientProperties webClientProperties) {
        this.webClientProperties = webClientProperties;
    }

    @Bean
    public WebClient webClient() {
        return WebClient.builder()
            .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
            .clientConnector(clientHttpConnector())
            .build();
    }

    private ReactorClientHttpConnector clientHttpConnector() {
        return new ReactorClientHttpConnector(
            options -> options.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, webClientProperties.getConnectTimeoutMs())
                .build());
    }
}

File: statistics-service/src/main/kotlin/com/vegesoft/moneytracker/statistics/configuration/WebClientConfiguration.kt
package com.vegesoft.moneytracker.statistics.configuration

import com.vegesoft.moneytracker.statistics.configuration.properties.WebClientProperties
import io.netty.channel.ChannelOption
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.http.HttpHeaders
import org.springframework.http.MediaType
import org.springframework.http.client.reactive.ReactorClientHttpConnector
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class WebClientConfiguration(private val webClientProperties: WebClientProperties) {
    @Bean
    fun webClient(): WebClient {
        return WebClient.builder()
                .defaultHeader(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE)
                .clientConnector(clientHttpConnector())
                .build()
    }

    private fun clientHttpConnector(): ReactorClientHttpConnector {
        return ReactorClientHttpConnector { options ->
            options.option(ChannelOption.CONNECT_TIMEOUT_MILLIS, webClientProperties.connectTimeoutMs)
                    .build()
        }
    }
}